### PR TITLE
Feature 19: Update only affected weekdays

### LINKED ISSUE #19: Updating order of activities when reordering in the weekplan
**Is your feature request related to a problem? Please describe.**
When placing an activity in the middle of a weekday, all activites below will have to be pushed down by updating their order. Currently, this requires the entire weekplan to be rerendered. 

**Describe the solution you'd like**
Only the activities affected should be updated.

**Describe alternatives you've considered**
Include a new endpoint in the activity controller, which takes an array of key-value pairs (key identifying the activity, value indicating the new order value) and updates the values in the database.
Alternatively, use decimal numbers for ordering activities and use a formula to place the reordered activity between two activities in the new location. This would make it unnecessary to update all activities below it, and might be able to use the existing update activity endpoint from issue #18 .

**Additional context**

### PR DESCRIPTION
**Pull request by AdrianPlesner**
Closes #19 
Still missing some tests

### CODE REVIEW

**Comment by Determinatoro** [comment]
Please use the following checklist, @13uller. Feel free to check it off as you go along.

**Code Design** 

- [ ] The code is in the right place? (Both in terms of folder structure and class structure)
- [ ] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems     
- [ ] This code could not, to the best of my knowledge, have reused existing code
- [ ] The code does not introduce functionality that is unnecessary for solving the problem

**Code Readability** 

- [ ] Names are meaningful and self-documenting
- [ ] It is understandable, by reading the code, what it does
- [ ] The code is simple and readable. i.e. it contains no \"hack\", or obscure solution   

**Code Maintainability**

- [ ] Has all the added code been properly commented

Only check one of the items below:

- [ ] The functionality does **not** need to be covered by tests
- [ ] The functionality needs to be covered by tests

Only check these if the functionality needs to be covered by tests:

- [ ] All functionality is covered by tests
- [ ] The tests has sensible names
- [ ] By reading the tests you know what they cover
- [ ] The tests cover sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing

**Functionality**

For this part, run the web-api locally as described in https://aau-giraf.github.io/wiki/development/rest_api_development/BuildAndRunLocally/ 
- [ ] The web-api runs without issues
- [ ] The functionality that the code claims to implement, are actually implemented
- [ ] The integration tests runs successfully. Refer to https://github.com/aau-giraf/web-api/tree/master/GirafIntegrationTest for how to run the tests

**Comment by Determinatoro** [comment]
Please use the following checklist, @Riasalit. Feel free to check it off as you go along.

**Code Design** 

- [ ] The code is in the right place? (Both in terms of folder structure and class structure)
- [ ] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems     
- [ ] This code could not, to the best of my knowledge, have reused existing code
- [ ] The code does not introduce functionality that is unnecessary for solving the problem

**Code Readability** 

- [ ] Names are meaningful and self-documenting
- [ ] It is understandable, by reading the code, what it does
- [ ] The code is simple and readable. i.e. it contains no \"hack\", or obscure solution   

**Code Maintainability**

- [ ] Has all the added code been properly commented

Only check one of the items below:

- [ ] The functionality does **not** need to be covered by tests
- [ ] The functionality needs to be covered by tests

Only check these if the functionality needs to be covered by tests:

- [ ] All functionality is covered by tests
- [ ] The tests has sensible names
- [ ] By reading the tests you know what they cover
- [ ] The tests cover sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing

**Functionality**

For this part, run the web-api locally as described in https://aau-giraf.github.io/wiki/development/rest_api_development/BuildAndRunLocally/ 
- [ ] The web-api runs without issues
- [ ] The functionality that the code claims to implement, are actually implemented
- [ ] The integration tests runs successfully. Refer to https://github.com/aau-giraf/web-api/tree/master/GirafIntegrationTest for how to run the tests

**Review by ZohraAmini — DISMISSED** [review]

File: GirafRest/Controllers/WeekController.cs [hunk 1]
```diff
@@ -191,6 +191,68 @@ public async Task<ActionResult> ReadUsersWeekSchedule(string userId, int weekYea
             }));
         }
 
+        [HttpGet("{userId}/{weekYear}/{weekNumber}/{day}")]
+        [Authorize]
+        [ProducesResponseType(typeof(SuccessResponse<WeekdayDTO>), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult> GetWeekDay(string userId, int weekYear, int weekNumber, int day)
```
Comment (JeppeKLau) [inline GirafRest/Controllers/WeekController.cs:199]: Alle dele af softwaren skal dokumenteres.

File: GirafRest/Controllers/WeekController.cs [hunk 2]
```diff
@@ -234,6 +296,52 @@ public async Task<ActionResult> UpdateWeek(string userId, int weekYear, int week
             return Ok(new SuccessResponse<WeekDTO>(new WeekDTO(week)));
         }
 
+        [HttpPut("day/{userId}/{weekYear}/{weekNumber}")]
```
Comment (JeppeKLau) [inline GirafRest/Controllers/WeekController.cs:299]: Kan det ikke overvejes at bruge en patch metode i stedet for put?

File: GirafRest/Controllers/WeekController.cs [hunk 3]
```diff
@@ -234,6 +296,52 @@ public async Task<ActionResult> UpdateWeek(string userId, int weekYear, int week
...
+            
+            Weekday newDay = new Weekday(weekdayDto);
+
+            newDay.Activities.Clear();
+            if (!await AddPictogramsToWeekday(newDay, weekdayDto, _giraf))
+            {
+                return NotFound(new ErrorResponse(ErrorCode.ResourceNotFound, "Missing pictogram"));
+            }
+
+            var oldDay = week.Weekdays.IndexOf( week.Weekdays.Single(d => d.Day == newDay.Day));
```
Comment (JeppeKLau) [inline GirafRest/Controllers/WeekController.cs:335]: Man må kunne give `oldDay` en type i stedet for `var`

**Review by JeppeKLau — DISMISSED** [review]

**Comment by JeppeKLau** [comment]
Der mangler også tests af `UpdateWeekday`

**Review by ZohraAmini — APPROVED** [review]

**Comment by Riasalit** [comment]
sååå, ved ikke om de ting den klager over i PictogramController.cs  og StatusController.cs er et problem?

**Review by Riasalit — APPROVED** [review]

**Pull request merged**